iris_grib/__init__.py

```diff
@@ -342,56 +342,23 @@ def _compute_extra_keys(self):
         self.extra_keys['_forecastTimeUnit'] = self._timeunit_string()
 
         # shape of the earth
-
-        # pre-defined sphere
-        if self.shapeOfTheEarth == 0:
+        soe_code = self.shapeOfTheEarth
+        # As this class is now *only* for GRIB1, 'shapeOfTheEarth' is not a value read from the actual file :  It is
+        # really a GRIB2 param, and the value is merely what eccodes (griabpi) considers the correct default.
+        # This was always =6 until eccodes 0.19, when it changed to 0.  See https://jira.ecmwf.int/browse/ECC-811
+        # The two represent different sized spherical earths.
+        if soe_code not in (6, 0):
+            raise ValueError('Unexpected shapeOfTheEarth value =', soe_code)
+        # *FOR NOW* maintain the old behaviour (radius=6371229) in all cases for backwards compatibility.
+        # However, this does not match 'radiusOfTheEarth' and may be incorrect :  We may change it in future.
+        soe_code = 6
+        if soe_code == 0:
+            # New supposedly-correct correct value, matches the 'radiusOfTheEarth' parameter.
```

Comment:
Is this intentional, you are hardcoding `soe_code` to be 6, but including the `if soe_code == 0` for future, if we decide to change it?

Reply:
Yes, I thought I should remove most of the previous unused options, but I left both these ones in for reference.